### ISSUE TITLE
[BUG] Azure data table client listEntities API return string instead of binary data (breaking change & bug) 

### ISSUE THREAD

**Issue by DragonLi-Mi** [Azure/azure-sdk-for-java#40385]
**Describe the bug**
Azure data table client listEntities API started to have a different behavior with getEntity API. And the change start from 12.4.0. It is same result in 12.3.19. 
**To Reproduce**
This is my data in azure data table 
![image](https://github.com/Azure/azure-sdk-for-java/assets/4321063/43161f40-ba82-47a9-b063-d451a7946b38)
 When I upgarde to 12.4.0. The ListEntityes API start to return binary data as String type, and break all my service. 

```java 

package org.example;
import com.azure.core.http.rest.PagedIterable;
import com.azure.core.util.Context;
import com.azure.data.tables.TableClient;
import com.azure.data.tables.TableClientBuilder;
import com.azure.data.tables.models.*;

import java.time.Duration;
import java.util.ArrayList;
import java.util.stream.Stream;

public class Main {
    public static void main(String[] args) {
        String connectionString = "";

        TableClient tableClient = new TableClientBuilder()
                .connectionString(connectionString)
                .tableName("ConfigurationManagementGraphSP")
                .buildClient();

        ListEntitiesOptions listEntitiesOptions = new ListEntitiesOptions()
                .setFilter("(PartitionKey eq '636f6e66696775726174696f6e') and RowKey eq '67726170682e6a616e757367726170682d76657273696f6e'");
        PagedIterable<TableEntity> tableEntities  = tableClient.listEntities(listEntitiesOptions, Duration.ofSeconds(60), Context.NONE);
        Stream<TableEntity> result = tableEntities.stream();
        ArrayList<TableEntity> tableEntityList = new ArrayList<>();
        result.forEach(tableEntityList::add);

        TableEntity retrievedEntity = tableClient.getEntity("636f6e66696775726174696f6e", "67726170682e6a616e757367726170682d76657273696f6e");
        Object data = retrievedEntity.getProperty("value");
    }
}

```

![image](https://github.com/Azure/azure-sdk-for-java/assets/4321063/61a98cd3-d186-4a24-81fc-5e85b7b58444)

**Expected behavior**
List entities API should correct return binary data as binary. 

**Setup (please complete the following information):**
 - OS: [Windows]
 - IDE: [IntelliJ]
 - Library/Libraries: [azure-data-tables:12.4.0]
 - Java version: [11]

**Comment by jairmyree** [comment]
@DragonLi-Mi Thank you for reaching out. 

This is a known Tables issue and a fix for this is currently being worked on. 
Thank you very much for your patience, I will update this issue when the fix has been merged.

**Comment by jairmyree** [comment]
@DragonLi-Mi A fix for this issue has been included in azure-data-table 12.4.2. Please update to the latest SDK version. Thank you for reporting this issue to us.
